src/installer/corehost/cli/apphost/static/CMakeLists.txt

```diff
@@ -78,62 +79,131 @@ if(CLR_CMAKE_TARGET_WIN32)
         ${CORECLR_STATIC_LIB_LOCATION}/libcoreclrpal.a
         ${CORECLR_STATIC_LIB_LOCATION}/libpalrt.a
         ${CORECLR_STATIC_LIB_LOCATION}/libeventprovider.a
         ${CORECLR_STATIC_LIB_LOCATION}/libnativeresourcestring.a
     )
-endif(CLR_CMAKE_TARGET_WIN32)
 
-if(CLR_CMAKE_TARGET_OSX)
-  find_library(COREFOUNDATION CoreFoundation)
-  find_library(CORESERVICES CoreServices)
-  find_library(SECURITY Security)
-  find_library(SYSTEM System)
-
-  LIST(APPEND CORECLR_LIBRARIES
-    ${COREFOUNDATION}
-    ${CORESERVICES}
-    ${SECURITY}
-    ${SYSTEM}
-  )
-endif(CLR_CMAKE_TARGET_OSX)
-
-# On OSX and *BSD, we use the libunwind that's part of the OS
-if(CLR_CMAKE_TARGET_FREEBSD)
-    find_unwind_libs(UNWIND_LIBS)
-
-    LIST(APPEND CORECLR_LIBRARIES
-      ${UNWIND_LIBS}
-    )
-endif(CLR_CMAKE_TARGET_FREEBSD)
+    # currently linking coreclr into the singlefilehost is only supported on linux
+    # the following code here would be needed if/when BSD and OSX are supported too
+    if(CLR_CMAKE_TARGET_OSX)
+      find_library(COREFOUNDATION CoreFoundation)
+      find_library(CORESERVICES CoreServices)
+      find_library(SECURITY Security)
+      find_library(SYSTEM System)
+
+      LIST(APPEND CORECLR_LIBRARIES
+        ${COREFOUNDATION}
+        ${CORESERVICES}
+        ${SECURITY}
+        ${SYSTEM}
+      )
+    endif(CLR_CMAKE_TARGET_OSX)
+
+    # On OSX and *BSD, we use the libunwind that's part of the OS
+    if(CLR_CMAKE_TARGET_FREEBSD)
+        find_unwind_libs(UNWIND_LIBS)
+
+        LIST(APPEND CORECLR_LIBRARIES
+          ${UNWIND_LIBS}
+        )
+    endif(CLR_CMAKE_TARGET_FREEBSD)
+
+    if(CLR_CMAKE_TARGET_NETBSD)
+        find_library(KVM kvm)
+
+        LIST(APPEND CORECLR_LIBRARIES
+          ${KVM}
+        )
+    endif(CLR_CMAKE_TARGET_NETBSD)
+endif(CLR_CMAKE_TARGET_WIN32)
 
-if(CLR_CMAKE_TARGET_NETBSD)
-    find_library(KVM kvm)
+# Path like: artifacts/bin/native/net5.0-Linux-Release-arm/
+set(NATIVE_LIBS_LOCATION "${NATIVE_LIBS_ARTIFACTS}")
+message ("Looking for native libs at location: '${NATIVE_LIBS_LOCATION}'.")
 
-    LIST(APPEND CORECLR_LIBRARIES
-      ${KVM}
+if(NOT CLR_CMAKE_TARGET_LINUX)
+    set(NATIVE_LIBS
+        # Native libs linked into singlefilehost is supported only on Linux for now.
+    )
+else()
+    set(NATIVE_LIBS
+      ${NATIVE_LIBS_LOCATION}/libSystem.IO.Compression.Native.a
+      ${NATIVE_LIBS_LOCATION}/libSystem.Native.a
+      ${NATIVE_LIBS_LOCATION}/libSystem.Net.Security.Native.a
+      ${NATIVE_LIBS_LOCATION}/libSystem.Security.Cryptography.Native.OpenSsl.a
     )
-endif(CLR_CMAKE_TARGET_NETBSD)
 
-if(NOT CLR_CMAKE_TARGET_WIN32)
-    set(DEF_SOURCES ${CMAKE_CURRENT_SOURCE_DIR}/singlefilehost_unixexports.src)
-    set(EXPORTS_FILE ${CMAKE_CURRENT_BINARY_DIR}/singlefilehost.exports)
-    generate_exports_file(${DEF_SOURCES} ${EXPORTS_FILE})
+    if(CLR_CMAKE_TARGET_OSX)
+        LIST(APPEND NATIVE_LIBS
+          ${NATIVE_LIBS_LOCATION}/libSystem.Security.Cryptography.Native.Apple.a
```

Comment:
Why are we adding this?

Reply:
We would be adding this on OSX, since OSX has one more native library.

Reply:
But that should not be part of this PR as we don't add any other static libraries for OSX, right? That's what I was asking about.

Reply:
I had two options - remove the non-linux code completely or keep it somehow, but not run. There was a suggestion above to comment it out and put explanation why (we kind of have an explanation already). 
Would that be better?

Reply:
Also, this cmake code path is now Linux only.

Reply:
I will comment out the non-Linux code. It seems indeed confusing too leave it as this.

Reply:
I would prefer just completely removing this whole if. It is very confusing and after we enable OSX, it would not be correct either (the libSystem.Security.Cryptography.Native.OpenSsl.a is not for OSX).

Reply:
Ok. I will remove this part.

Reply:
Pushed a change. Please take a look if it is what you expected.